### PR TITLE
Dataplane work

### PR DESCRIPTION
**Pull request by rabi**
Initial patches to required to integrate with dataplane-operator.

### CODE REVIEW

File: api/v1beta1/openstackbaremetalset_types.go [hunk 1]
```diff
@@ -29,7 +39,15 @@ type OpenStackBaremetalSetSpec struct {
...
 	RhelImageURL string `json:"rhelImageUrl,omitempty"`
+
+	// When set to disabled, automated cleaning will be avoided
+	// during provisioning and deprovisioning.
+	// +optional
+	// +kubebuilder:default:=metadata
 	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
+
+	// +kubebuilder:validation:OptionIronicAutomatedCleanal
```
Comment (abays) [inline api/v1beta1/openstackbaremetalset_types.go:50]: This looks like a typo

File: api/v1beta1/openstackbaremetalset_types.go [hunk 2]
```diff
@@ -29,7 +39,15 @@ type OpenStackBaremetalSetSpec struct {
 	BaremetalHosts map[string]string `json:"baremetalHosts"`
 	// RhelImageURL - Remote URL pointing to desired RHEL qcow2 image
 	RhelImageURL string `json:"rhelImageUrl,omitempty"`
+
+	// When set to disabled, automated cleaning will be avoided
+	// during provisioning and deprovisioning.
+	// +optional
+	// +kubebuilder:default:=metadata
```
Comment (abays) [inline api/v1beta1/openstackbaremetalset_types.go:46]: Might be an extra `:` after `default`

Comment (rabi) [reply api/v1beta1/openstackbaremetalset_types.go]: The doc asks for it seems. Anyway I did not see it anywhere used, so changed. https://book.kubebuilder.io/reference/markers/crd-validation.html

File: api/v1beta1/openstackbaremetalset_types.go (hunk 1 above)
Comment (rabi) [reply api/v1beta1/openstackbaremetalset_types.go]: My bad.

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**